benchmarks/src/main/scala/benchmark/MergeBenchmark.scala

```diff
@@ -0,0 +1,145 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package benchmark
+
+import java.util.UUID
+
+import org.apache.spark.sql.Row
+
+trait MergeConf extends BenchmarkConf {
+  def scaleInGB: Int
+  def tableName: String = "web_returns"
+  def userDefinedDbName: Option[String]
+  def dbName: String = userDefinedDbName.getOrElse(s"merge_sf${scaleInGB}")
+  def dbLocation: String = dbLocation(dbName)
+}
+
+case class MergeBenchmarkConf(
+     scaleInGB: Int = 0,
+     userDefinedDbName: Option[String] = None,
+     iterations: Int = 3,
+     benchmarkPath: Option[String] = None) extends MergeConf {
+}
+
+object MergeBenchmarkConf {
+  import scopt.OParser
+  private val builder = OParser.builder[MergeBenchmarkConf]
+  private val argParser = {
+    import builder._
+    OParser.sequence(
+      programName("Merge Benchmark"),
+      opt[String]("scale-in-gb")
+        .required()
+        .valueName("<scale of benchmark in GBs>")
+        .action((x, c) => c.copy(scaleInGB = x.toInt))
+        .text("Scale factor in GBs of the TPCDS benchmark"),
+      opt[String]("benchmark-path")
+        .required()
+        .valueName("<cloud storage path>")
+        .action((x, c) => c.copy(benchmarkPath = Some(x)))
+        .text("Cloud path to be used for creating table and generating reports"),
+      opt[String]("iterations")
+        .optional()
+        .valueName("<number of iterations>")
+        .action((x, c) => c.copy(iterations = x.toInt))
+        .text("Number of times to run the queries"))
+  }
+
+  def parse(args: Array[String]): Option[MergeBenchmarkConf] = {
+    OParser.parse(argParser, args, MergeBenchmarkConf())
+  }
+}
+
+class MergeBenchmark(conf: MergeBenchmarkConf) extends Benchmark(conf) {
+  /**
+   * Runs every merge test case multiple times and records the duration.
+   */
+  override def runInternal(): Unit = {
+    for ((k, v) <- extraConfs) spark.conf.set(k, v)
+    spark.sparkContext.setLogLevel("WARN")
+    log("All configs:\n\t" + spark.conf.getAll.toSeq.sortBy(_._1).mkString("\n\t"))
+    spark.sql(s"USE ${conf.dbName}")
+
+    val targetRowCount = spark.read.table(s"`${conf.dbName}`.`target_${conf.tableName}`").count
+
+    for (iteration <- 1 to conf.iterations) {
+      MergeTestCases.testCases.foreach { runMerge(_, targetRowCount, iteration = Some(iteration)) }
+    }
+    val results = getQueryResults().filter(_.name.startsWith("q"))
+    if (results.forall(x => x.errorMsg.isEmpty && x.durationMs.nonEmpty) ) {
+      val medianDurationSecPerQuery = results.groupBy(_.name).map { case (q, results) =>
+        assert(results.size == conf.iterations)
+        val medianMs = results.map(_.durationMs.get).sorted
+            .drop(math.floor(conf.iterations / 2.0).toInt).head
```

Comment:
To be fair, when data points size is even, the median is computed as the average of the two median points.

Reply:
I'm now relying on Utils.median instead of a custom implementation.